### PR TITLE
DBZ-2388 Expand use of Apicurio Converters in Debezium Server

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-2388

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Some testing would be nice here.

**Comment by gunnarmorling** [comment]
Hey @ani-sha, where are we with this one?

**Comment by ani-sha** [comment]
Hey @gunnarmorling, I am still working on this, quite delayed on this one!
Turns out there are other blockers in this one. As you suggested on creating `ApicurioTestResourceLifecycleManager` and proceeding which starts apicurio well, but I figured out as `debezium.format.key/value` is not set anywhere in `TestConfigSource` ie `DebeziumServer` by default uses `JsonConverter` [here](https://github.com/ani-sha/debezium/blob/3d02eb47ac48312e8f67129ded1920e5b91963f8/debezium-server/debezium-server-core/src/main/java/io/debezium/server/DebeziumServer.java#L160).

If I specify the format as `Avro` in `TestConfigSource` and adding required props (apicurio.registry.url, apicurio.registry.global-id) in `DebeziumServer` it uses `io.apicurio.registry.utils.converter.AvroConverter` for `DebeziumServerWithApicurioRegistryIT` but there remains connection failure. Also this causes failure for other ITs because of the same config properties. 

```
integrationTest.put("debezium.format.key", Avro.class.getSimpleName().toLowerCase());        
```
If this is too complicated for you to understand I can push the changes in a test branch and reference it. If you think of any other way to approach this do let me know. Thanks.

File: debezium-server/debezium-server-core/src/test/java/io/debezium/server/TestConfigSource.java [hunk 1]
```diff
@@ -38,6 +38,9 @@ public TestConfigSource() {
         integrationTest.put("debezium.source.schema.include.list", "inventory");
         integrationTest.put("debezium.source.table.include.list", "inventory.customers");
 
+        // integrationTest.put("debezium.format.key", Avro.class.getSimpleName().toLowerCase());
+        // integrationTest.put("debezium.format.value", Avro.class.getSimpleName().toLowerCase());
```
Comment (ani-sha) [inline debezium-server/debezium-server-core/src/test/java/io/debezium/server/TestConfigSource.java:42]: uncomment this part to use `debezium.format` as `avro`.

**Comment by ani-sha** [comment]
@gunnarmorling as per the discussion pushed the work in progress, take a look whenever you are free, thanks.

**Comment by gunnarmorling** [comment]
@ani-sha, so where are we with this one (sorry, I forgot about the details since we last spoke about it)?

**Comment by ani-sha** [comment]
hey @gunnarmorling, as per our discussion we intented to use `debezium.format.key/value` as `avro` but we struggled to start the apicurio connection, hence you asked to push  the changes to have a look. I didn't seem to figure out the reason for failed connection even using the mapped port. also, there were concerns regarding how/where to pass the the key/value into `DebeziumServer` without making it apicurio dependent.

**Comment by gunnarmorling** [comment]
@ani-sha, so re this one, seems you'd need some help to get unstuck?

**Comment by ani-sha** [comment]
@gunnarmorling Some help would be great especially for establishing a connection to start Apicurio using the Lifecycle Manager.

**Comment by jpechane** [comment]
@ani-sha Well, the test works if you add `-Dtest.type=IT`. Then messages arrives as binary encoded values expected. Unfortuntaely there is another problem.
IIRC the queskus application is started only once at the beginning of test suite execution. It is thus not possible to start DebeziumServer twice in the same Maven module with different converter configuration.
Maybe we would need to introduce a separate Maven module where all apicurio testing will be concentrated?
@gunnarmorling Any thought on the above?

**Comment by gunnarmorling** [comment]
@jpechane, ah yes, that's a good point. Perhaps two Maven profiles with different configurations would do the trick, and CI would run both of them, one after the other?

**Comment by jpechane** [comment]
@gunnarmorling Good idea. Then we will even not need a separate test. The only difference would be assertion rules for the content. In this regarde we can take inspiration from openshit-tests that runs once and there is assertion strtaegy different for each converter.

**Comment by ani-sha** [comment]
Thanks for the suggestions, working on it in a new branch, closing this PR.

**Pull request closed without merging**